libcxx/test/std/ranges/range.adaptors/range.stride.view/adaptor.pass.cpp

```diff
@@ -0,0 +1,133 @@
+        auto strided_iter                                                   = strided.begin();
+
+        // Check that the begin() iter views arr[0]
+        assert(*strided_iter == arr[0]);
+
+        // Check that the strided_iter, after advancing it 2 * 1 steps, views arr[2].
+        std::ranges::advance(strided_iter, 2);
+        assert(*strided_iter == arr[2]);
+      }
+      {
+        using View = InputView<bidirectional_iterator<int*>>;
+        View view(bidirectional_iterator<int*>(arr), bidirectional_iterator<int*>(arr + array_n));
+        std::same_as<std::ranges::stride_view<View>> decltype(auto) strided = view | std::views::stride(2);
+        auto strided_iter                                                   = strided.begin();
+
+        assert(*strided_iter == arr[0]);
+
+        // Same test as above, just advance one time with a bigger step (1 * 2 steps).
+        std::ranges::advance(strided_iter, 1);
+        assert(*strided_iter == arr[2]);
+      }
+    }
+  }
+
+  // Check various forms of
+  // adaptor | stride
+  {
+    // Parallels the two tests from above.
+    constexpr const auto identity_lambda = [](const int i) { return i * 2; };
+    {
+      using View = InputView<bidirectional_iterator<int*>>;
+      View view(bidirectional_iterator<int*>(arr), bidirectional_iterator<int*>(arr + array_n));
+      const auto transform_stride_partial = std::views::transform(identity_lambda) | std::views::stride(1);
+
+      auto transform_stride_applied      = transform_stride_partial(view);
+      auto transform_stride_applied_iter = transform_stride_applied.begin();
+      assert(*transform_stride_applied_iter == std::invoke(identity_lambda, arr[0]));
+      std::ranges::advance(transform_stride_applied_iter, 2);
+      assert(*transform_stride_applied_iter == std::invoke(identity_lambda, arr[2]));
+    }
+
+    {
+      using View = InputView<bidirectional_iterator<int*>>;
+      View view(bidirectional_iterator<int*>(arr), bidirectional_iterator<int*>(arr + array_n));
+      const auto transform_stride_partial = std::views::transform(identity_lambda) | std::views::stride(2);
+
+      const auto transform_stride_applied = transform_stride_partial(view);
+      auto transform_stride_applied_iter  = transform_stride_applied.begin();
+      assert(*transform_stride_applied_iter == std::invoke(identity_lambda, arr[0]));
+      std::ranges::advance(transform_stride_applied_iter, 1);
+      assert(*transform_stride_applied_iter == std::invoke(identity_lambda, arr[2]));
+    }
+  }
+
+  {
+    using ForwardStrideView      = std::ranges::stride_view<InputView<forward_iterator<int*>>>;
+    using BidirStrideView        = std::ranges::stride_view<InputView<bidirectional_iterator<int*>>>;
+    using RandomAccessStrideView = std::ranges::stride_view<InputView<random_access_iterator<int*>>>;
+
+    static_assert(std::ranges::forward_range<ForwardStrideView>);
```

Comment:
We usually test constraints and concepts in the global namespace, before all other tests.

Reply:
I believe that this is addressed in a96dd45cbad8b7c0d6e6a2e3fa008471be88ef12.